### DISCUSSION TITLE
Add tcp option

### DISCUSSION THREAD

**Discussion by ehsan-elwan** [kokorin/Jaffree#313] [Q&A] [answered]
I have an ffmpeg that forward a local udp stream to a remote server using this command:
`ffmpeg -loglevel level+warning -n -progress tcp://127.0.0.1:62012 -fflags +genpts -fflags +nofillin -fflags +discardcorrupt -i udp://127.0.0.1:1651?reuse=1&overrun_nonfatal=1&fifo_size=50000000&buffer_size=20000000 -c:v copy -c:a copy -f flv rtmp://stream-source.abc.com:1935/live/streamcam10`

I'm able to generate the above command with the ffmpeg wrapper, but the command fail due to TCP negotiation failed

I found that I need to add a tcp option to make it work like so:
`ffmpeg -loglevel level+warning -n -progress tcp://127.0.0.1:62012\?listen=2`

The above command works but I'm trying to away to add the `listen` option to tcp but I can't find away to set it up in the wrapper any help is much appreciated

**Comment by kokorin** [comment, marked as answer]
Probably, you specified `?listen=2` at wrong option. `-progress` tells ffmpeg where to send progress reports, and this port is opened by Jaffree itself.
Please, provide code sample and logs. Issue template clearly says that it should be provided.
